main.go

```diff
@@ -25,86 +47,162 @@ func getEnv(key, fallback string) string {
 	return fallback
 }
 
+var versionedExpressions = map[string]expression{
+	"0.11": expression{
+		planStatusRegex: regexp.MustCompile(
+			"^(.*?): (.*?) +\\(ID: (.*?)\\)$",
+		),
+		reTfPlanLine: regexp.MustCompile(
+			"^( +)([a-zA-Z0-9%._-]+):( +)([\"<])(.*?)([>\"]) +=> +([\"<])(.*)([>\"])(.*)$",
+		),
+		reTfPlanCurrentResource: regexp.MustCompile(
+			"^([~/+-]+) (.*?) +(.*)$",
+		),
+		resourceIndex: 2,
+		assign:        ":",
+		operator:      "=>",
+	},
+	"0.12": expression{
+		planStatusRegex: regexp.MustCompile(
+			"^(.*?): (.*?) +\\[id=(.*?)\\]$",
+		),
+		reTfPlanLine: regexp.MustCompile(
+			"^( +)([ ~a-zA-Z0-9%._-]+)=( +)([\"<])(.*?)([>\"]) +-> +(\\()(.*)(\\))(.*)$",
+		),
+		reTfPlanCurrentResource: regexp.MustCompile(
+			"^([~/+-]+) (.*?) +(.*) (.*) (.*)$",
+		),
+		resourceIndex: 3,
+		assign:        "=",
+		operator:      "->",
+	},
+}
+
 func main() {
 	log.SetFlags(0) // no timestamps on our logs
 
 	// Character used to mask sensitive output
 	var tfmaskChar = getEnv("TFMASK_CHAR", "*")
-
 	// Pattern representing sensitive output
-	var tfmaskValuesRegex = getEnv("TFMASK_VALUES_REGEX", "(?i)^.*(oauth|secret|token|password|key|result).*$")
-
+	var tfmaskValuesRegex = getEnv("TFMASK_VALUES_REGEX",
+		"(?i)^.*(oauth|secret|token|password|key|result|id).*$")
 	// Pattern representing sensitive resource
-	var tfmaskResourceRegex = getEnv("TFMASK_RESOURCES_REGEX", "(?i)^(random_id).*$")
-
-	// stage.0.action.0.configuration.OAuthToken: "" => "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
-	reTfPlanLine := regexp.MustCompile("^( +)([a-zA-Z0-9%._-]+):( +)([\"<])(.*?)([>\"]) +=> +([\"<])(.*)([>\"])(.*)$")
-
-	// random_id.some_id: Refreshing state... (ID: itILf4x5lqleQV9ZwT2gH-Zg3yuXM8pdUu6VFTX...P5vqUmggDweOoxFMPY5t9thA0SJE2EZIhcHbsQ)
-	reTfPlanStatusLine := regexp.MustCompile("^(.*?): (.*?) +\\(ID: (.*?)\\)$")
-	
+	var tfmaskResourceRegex = getEnv("TFMASK_RESOURCES_REGEX",
+		"(?i)^(random_id|random_string).*$")
 
-	// -/+ random_string.postgres_admin_password (tainted) (new resource required)
-	reTfPlanCurrentResource := regexp.MustCompile("^([~/+-]+) (.*?) +(.*)$")
-	reTfApplyCurrentResource := regexp.MustCompile("^([a-z].*?): (.*?)$")
-	currentResource := ""
+	// Default to tf 0.11, but users can override
+	var tfenv = getEnv("TFENV", "0.11")
```

Comment:
I think we should change the default to be what most users should be running rather than worry about breaking the current default behavior. Since we're in a 0.x release, it's fair to change that especially since there's an easy work around by exporting this env. 

```suggestion
	var tfenv = getEnv("TFENV", "0.12")
```